package.json

```diff
@@ -12,22 +12,32 @@
     "*.js"
   ],
   "scripts": {
-    "build": "tsc --project tsconfig.build.json",
+    "build": "pnpm run contracts:compile && tsc --project tsconfig.build.json",
     "clean": "rm -rf coverage dist",
+    "contracts:compile:force": "hardhat compile --force",
+    "contracts:compile": "hardhat compile",
+    "dev:eth-node": "hardhat node",
     "dev": "nodemon --ext ts,js,json,env  --exec \"pnpm ts-node src/index.ts\"",
+    "docker:build": "docker build -t api3/airseekerv2:latest -f docker/Dockerfile .",
+    "docker:run": "docker run -it --rm api3/airseekerv2:latest",
     "eslint:check": "eslint --report-unused-disable-directives --cache --ext js,ts . --max-warnings 0",
     "eslint:fix": "pnpm run eslint:check --fix",
     "prettier:check": "prettier --check \"./**/*.{js,ts,md,json}\"",
     "prettier:fix": "prettier --write \"./**/*.{js,ts,md,json}\"",
-    "test": "jest --config=jest-unit.config.js",
     "test:e2e": "jest ---config=jest-e2e.config.js --runInBand",
-    "tsc": "tsc --project .",
-    "docker:build": "docker build -t api3/airseekerv2:latest -f docker/Dockerfile .",
-    "docker:run": "docker run -it --rm api3/airseekerv2:latest",
-    "dev:eth-node": "hardhat node"
+    "test": "jest --config=jest-unit.config.js",
+    "tsc": "tsc --project ."
   },
   "devDependencies": {
     "@api3/ois": "^2.2.0",
+    "@ethersproject/abi": "^5.7.0",
```

Comment:
I had to manually install `@ethersproject/abi` and `@ethersproject/providers` because I was getting TS errors in Typechain generated files. This may be an issue with how pnpm internal dependency scoping works, but I haven't investigated deeply because this is dev only dependency and a temporary solution (the proper solution would be to import the Typechain artifacts from the published repo).

Reply:
Actually nevermind. There were no issues with pnpm, but a mismatch of ethers v6 and v5 in the hardhat dependencies.